Pod/Classes/CDTQUnindexedMatcher.m

```diff
@@ -178,24 +178,32 @@ - (BOOL)executeSelectorTree:(CDTQQueryNode *)node onRevision:(CDTDocumentRevisio
         BOOL invertResult = [operator isEqualToString:NOT];
         if (invertResult) {
             operatorExpression = operatorExpression[NOT];
-            operator= operatorExpression.allKeys[0];
+            operator = operatorExpression.allKeys[0];
         }
 
         NSObject *expected = operatorExpression[operator];
         NSObject *actual = [CDTQValueExtractor extractValueForFieldName:fieldName fromRevision:rev];
-
+        // Since $in is the same as a series of $eq comparisons -
+        // Treat them the same by:
+        // - Ensuring that both expected and actual are NSArrays.
+        // - Convert the $in operator to the $eq operator.
+        if (![expected isKindOfClass:[NSArray class]]) {
+            expected = @[ expected ];
+        }
+        if (![actual isKindOfClass:[NSArray class]]) {
+            actual = @[ actual ? : [NSNull null] ];
```

Comment:
I assume `actual ? : [NSNull null]` is the same as `actual ? actual : [NSNull null]`, but I'd rather you spelt it out so I didn't have to guess. The `?` operator has enough potential for confusion. Could you instead use `actual = actual ? @[ actual ] : @[]` which reduces the number of nested square brackets?